internal/utils/logger.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2022 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package utils
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+)
+
+var (
+	errorLogger *log.Logger
+	warnLogger  *log.Logger
+	infoLogger  *log.Logger
+	traceLogger *log.Logger
+)
+
+// Higher log level means more logging
+//
+func InitLogging(logLevel int) {
+	errorLogger = initLogger("ERROR", 0, logLevel)
+	warnLogger = initLogger("WARNING", 1, logLevel)
+	infoLogger = initLogger("INFO", 2, logLevel)
+	traceLogger = initLogger("TRACE", 3, logLevel)
+}
+
+func Trace(msg string, params ...any) {
+	logHelper(traceLogger, msg, params...)
+}
+
+func Warn(msg string, params ...any) {
+	logHelper(warnLogger, msg, params...)
+}
+
+func Info(msg string, params ...any) {
+	logHelper(infoLogger, msg, params...)
+}
+
+func Error(msg string, params ...any) {
+	logHelper(errorLogger, msg, params...)
+}
+
+// Not being used currently, but how do we feel about it..?
+// Do any finalizers or teardowns get skipped if we just exit like this?
+// Is it really useful to return err through 16 levels of call stack???
+//
+func Fatal(msg string, params ...any) {
+	logHelper(errorLogger, msg, params...)
+	os.Exit(1)
+}
```

Comment:
Could also have a Panic function here maybe.

Reply:
Well, there should only be errors from ec that surface to the end users, at each point we do `if err != nil` we need to introduce a specific enough error [code, message], and/or handle the `err` in that spot with specificity it requires. Errors then end up being one level deep. That is why [HACBS-633](https://issues.redhat.com/browse/HACBS-633) is important.
We might not want to exit with the same status for every case, we might not want any `os.Exit` calls other than the ones in the error handling logic.

Reply:
I also need to note here that the shed needs to be blue.

Reply:
FYI, in most cases, `panic` is not the answer: https://go.dev/doc/effective_go#panic